consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validation/ProposalValidator.java

```diff
@@ -93,6 +96,16 @@ public ProposalValidator(
    */
   public boolean validate(final Proposal msg) {
 
+    checkState(
+        protocolSchedule instanceof BftProtocolSchedule,
+        "Wrong class type for protocol schedule, requires BftProtocolSchedule");
+
+    final BlockValidator blockValidator =
```

Comment:
I think in this case you can also use the ProtocolSchedule.getByBlockHeader as we have the block header on the received proposal message block i.e. ```protocolSchedule.getByBlockHeader(                msg.getBlock().getHeader())```

Reply:
Done 👍